Sources/MapboxCoreNavigation/PassiveLocationManager.swift

```diff
@@ -236,7 +236,8 @@ open class PassiveLocationManager: NSObject {
             .rawLocationKey: rawLocation,
             .matchesKey: matches,
             .roadNameKey: status.roadName,
-            .routeShieldRepresentationKey: status.routeShieldRepresentation
+            .routeShieldRepresentationKey: status.routeShieldRepresentation,
+            .speedKey: max(lastLocation.speed, 0.0)
```

Comment:
If the speed is negative, should we be setting this key at all? A negative `CLLocationSpeed` is documented as meaning an invalid or unavailable speed, so what you have is valid. But we also have the ability to not set this key, resulting in a nil current speed in the UI. Either way, we should document the fact that the speed can be unavailable.

Reply:
We could handle the invalid speed data and filter it out later inside the `SpeedLimitView`. And in this case, we don't need to provide it through the `Notification`. I just removed the additional key for the speed in `Notification` and use the raw `CLLocationSpeed` for the `SpeedLimitView.currentSpeed`